### PR TITLE
Add the knative-publish app

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

- add the knative-publish app
- refactor the publish test helpers to be reusable for both the `publish` app and the `knative-publish` app
- add the tests for the knative-publish app
- add the Dockerfile for the knative-publish app
- add the prow build and release steps for the knative-publish app

**Related issue(s)**
#1662

### CODE REVIEW

**Comment by marcobebway** [comment]
The basic knative publish application logic is developed.
Next steps is to write more tests.

**Comment by marcobebway** [comment]
The basic knative publish application logic is developed.
The next step is to write more tests.

File: components/event-bus/cmd/event-bus-publish-knative/application/application_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+func Test_KnativePublishApplication_Started(t *testing.T) {
+	assert.NotNil(t, (*application).tracer)
+	assert.NotNil(t, (*application).serveMux)
+	assert.NotNil(t, (*application).publisher)
+	assert.NotNil(t, (*application).knativeLib)
+	assert.Equal(t, true, (*application).started)
+}
+
+// todo add more tests
```
Comment (marcobebway) [inline components/event-bus/cmd/event-bus-publish-knative/application/application_test.go]: Next step is to write more tests here.

**Comment by radufa** [comment]
Please move the "fake" package into a "test" directory, so that it will be not part of the production binary.
See: https://github.com/golang-standards/project-layout

File: components/event-bus/cmd/event-bus-publish-knative/validators/publish_validator.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		return nil, api.ErrorResponseBadRequest()
+	}
+
+	// validate read the request body
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("failed to read request body: %v", err)
+		return nil, api.ErrorResponseInternalServer()
+	}
+
```
Comment (radufa) [inline components/event-bus/cmd/event-bus-publish-knative/validators/publish_validator.go]: Please add here:
defer r.Body.Close()

Comment (marcobebway) [reply components/event-bus/cmd/event-bus-publish-knative/validators/publish_validator.go]: Done right after reading the body.

**Pull request merged**